dataprofiler/profilers/helpers/report_helpers.py

```diff
@@ -100,7 +100,68 @@ def _prepare_report(report, output_format=None, omit_keys=None):
         if isinstance(value, set):
             value = sorted(list(value))
 
-        if isinstance(value, dict):
+        if "*" in omit_keys:
+            continue
+
+        # For data_stats (in structured case), need to recurse through a list
+        elif key == "data_stats" and isinstance(value, list) \
+                and "data_stats" not in omit_keys:
+
+            # Map cols to keep in report with their layer's omit keys
+            kept_cols_oks = dict()
+            # Cols that will have None in report since they were omitted
+            none_cols = []
+            for col_ind in range(len(value)):
+                col_name = str(value[col_ind].get('column_name'))
+                # column is being omitted
+                if f"data_stats.{col_name}" \
+                        in omit_keys:
+                    none_cols.append(col_ind)
+                    continue
+
+                # update omit keys
+                next_layer_omit_keys = []
+                # Track omit_keys that pertain to data_stats
+                ds_omit_keys = [omit_key[11:] for omit_key in omit_keys
+                                if omit_key[:11] == "data_stats."]
+                for omit_key in ds_omit_keys:
+                    omit_key_split = omit_key.split('.', 1)
+
+                    # this omit occurs only at this level, don't include
+                    if len(omit_key_split) == 1 and omit_key in {"*", col_name}:
+                        none_cols.append(col_ind)
+                        continue
```

Comment:
instead of continue, should we actually break here since we identified it? 

Since the second half already requires it to be an exact match, I don't think we need the `len(omit_key_split) == 1`, no?